src/repository.rs

```diff
@@ -403,6 +411,42 @@ impl Repository {
     self.inner.set_head(&refname)?;
     Ok(())
   }
+
+  #[napi]
+  /// Extract a signature from an object identified by its ID.
+  ///
+  /// This method can be used for any object that may be signed, such as commits or tags.
+  ///
+  /// @category Repository/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Repository {
+  ///   extractSignature(oid: string): { signature: string | null, signedData: string } | null;
```

Comment:
Can we also change the return type here?
```suggestion
  ///   extractSignature(oid: string): ExtractedSignature | null;
```